src/gsCore/gsBoundary.h

```diff
@@ -428,12 +407,196 @@ struct patchCorner : public boxCorner
     }
 };
 
+// We do not need comparization operators for boxCorner since it decays to a number anyway
+
+inline bool operator== (const patchCorner& a, const patchCorner& b)
+{ return a.patch == b.patch && a.m_index == b.m_index; }
+inline bool operator<  (const patchCorner& a, const patchCorner& b)
+{ return a.patch < b.patch || (a.patch == b.patch && a.m_index < b.m_index); }
+GISMO_DELEGATING_COMPARIZATION_OPERATORS(patchCorner)
+
+// This might cause terrible bugs
+GISMO_DELETE_COMPARIZATION_OPERATORS(boxCorner,patchCorner)
+
+/**
+ *   @brief Struct which represents a certain component (interior, face, egde, corner).
+ *
+ *   This struct is a generalization of \a boxSide and of \a boxCorner.
+ *
+ */
+
+struct GISMO_EXPORT boxComponent {
+private:
+    index_t m_index;        ///< The index defines the component
```

Comment:
Can you add short example of the values of the m_index as a comment ?
How does it correspond in 2D ? a couple of instances in 3D ?

Reply:
done